src/main/java/io/orangebeard/listener/v3client/v3Client.java

```diff
@@ -0,0 +1,164 @@
+    private final String projectName;
+    private boolean connectionWithOrangebeardIsValid;
+
+    protected final UUID uuid;
+
+
+    protected HttpHeaders getAuthorizationHeaders(String accessToken) {
+        HttpHeaders headers = new HttpHeaders();
+        headers.setBearerAuth((accessToken));
+        headers.setContentType(APPLICATION_JSON);
+        return headers;
+    }
+
+    public v3Client(String endpoint, UUID uuid, String projectName, boolean connectionWithOrangebeardIsValid) {
+        this.uuid = uuid;
+        SimpleClientHttpRequestFactory factory = new SimpleClientHttpRequestFactory();
+        factory.setConnectTimeout(30000);
+        this.restTemplate = new RestTemplate(factory);
+        this.endpoint = endpoint;
+        this.projectName = projectName == null ? null : projectName.toLowerCase();
+        this.connectionWithOrangebeardIsValid = connectionWithOrangebeardIsValid;
+    }
+
+    public v3Client(RestTemplate restTemplate, String endpoint, UUID uuid, String projectName, boolean connectionWithOrangebeardIsValid) {
+        this.uuid = uuid;
+        this.restTemplate = restTemplate;
+        this.endpoint = endpoint;
+        this.projectName = projectName == null ? null : projectName.toLowerCase();
+        this.connectionWithOrangebeardIsValid = connectionWithOrangebeardIsValid;
+    }
+
+    public UUID startTestRun(StartTestRun testRun) {
+        if (this.connectionWithOrangebeardIsValid) {
+            try {
+                HttpEntity<StartTestRun> request = new HttpEntity(testRun, this.getAuthorizationHeaders(this.uuid.toString()));
+                return this.restTemplate.exchange(
+                        String.format("%s/listener/v3/%s/test-run/start", this.endpoint, this.projectName),
+                        HttpMethod.POST,
+                        request,
+                        TestRunUUID.class).getBody().getTestRunUUID();
+
+            } catch (Exception var3) {
+                LOGGER.error("The connection with Orangebeard could not be established! Check the properties and try again!");
+                this.connectionWithOrangebeardIsValid = false;
+            }
+        }
+        return null;
+    }
+
+    public List<Suite> startSuite(StartSuiteRQ suiteRQ) {
+        if (this.connectionWithOrangebeardIsValid) {
+            HttpEntity<FinishTestItem> request = new HttpEntity(suiteRQ, this.getAuthorizationHeaders(this.uuid.toString()));
+            ResponseEntity<Suite[]> suite = this.restTemplate.exchange(
+                    String.format("%s/listener/v3/%s/suite/start", this.endpoint, this.projectName),
+                    HttpMethod.POST,
+                    request,
+                    Suite[].class);
+
+            return Arrays.asList(suite.getBody());
+        } else {
+            LOGGER.warn("The connection with Orangebeard could not be established!");
+            return null;
+        }
+    }
+
+    public UUID startTestItem(StartTest testItem) {
```

Comment:
TestItem is old domain model. We're migrating to the new domain model. We're starting tests, not test items.